controlplane/internal/controllers/scale.go

```diff
@@ -350,10 +348,10 @@ func (r *RKE2ControlPlaneReconciler) generateRKE2Config(ctx context.Context, rcp
 	return bootstrapRef, nil
 }
 
-func (r *RKE2ControlPlaneReconciler) generateMachine(ctx context.Context, rcp *controlplanev1.RKE2ControlPlane, cluster *clusterv1.Cluster, infraRef, bootstrapRef *corev1.ObjectReference) error {
-	newVersion, err := bsutil.Rke2ToKubeVersion(rcp.Spec.Version)
+func (r *RKE2ControlPlaneReconciler) generateMachine(ctx context.Context, rcp *controlplanev1.RKE2ControlPlane, cluster *clusterv1.Cluster, infraRef, bootstrapRef *corev1.ObjectReference, failureDomain *string) error {
+	newVersion, err := bsutil.Rke2ToKubeVersion(rcp.Spec.AgentConfig.Version)
```

Comment:
Need to check the error before doing the log

Reply:
moved the error check up and wrapped the error because it comes from an external package.